File: app/src/main/java/com/superproductivity/superproductivity/TaskListWidgetViewsService.java
package com.superproductivity.superproductivity;

import android.content.Intent;
import android.util.Log;
import android.widget.RemoteViewsService;

public class TaskListWidgetViewsService extends RemoteViewsService {
    @Override
    public RemoteViewsFactory onGetViewFactory(Intent intent) {
        Log.v("TW", "onGetViewFactory");
        return new TaskListWidgetViewsFactory(this, intent);
    }
}

File: app/src/main/java/com/superproductivity/superproductivity/TaskListWidgetViewsService.kt
package com.superproductivity.superproductivity

import android.content.Intent
import android.util.Log
import android.widget.RemoteViewsService

class TaskListWidgetViewsService : RemoteViewsService() {

    override fun onGetViewFactory(intent: Intent?): RemoteViewsFactory {
        Log.v("TW", "onGetViewFactory")
        return TaskListWidgetViewsFactory(this)
    }
}